src/reports/data/sonata_data.cpp

```diff
@@ -0,0 +1,276 @@
+#include <algorithm>
+#include <iostream>
+#include <memory>
+
+#include "../io/hdf5_writer.h"
+#include "../library/implementation_interface.hpp"
+#include "../library/sonatareport.h"
+#include "sonata_data.h"
+
+namespace bbp {
+namespace sonata {
+
+SonataData::SonataData(const std::string& report_name,
+                       const std::string& population_name,
+                       size_t max_buffer_size,
+                       int num_steps,
+                       double dt,
+                       double tstart,
+                       double tend,
+                       std::shared_ptr<nodes_t> nodes)
```

Comment:
It is just increasing the ref count by one.

Reply:
But that's unnecessary, no? as per `either use std::move or a reference`?

Reply:
In theory I could pass it as a const reference, because the report object should live longer than the sonata_data one. But I thought that the penalty was minimal.
see https://stackoverflow.com/questions/8385457/should-i-pass-a-shared-ptr-by-reference